front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/RedBlackToBlueGreenK8sPipelinesMigration.java

```diff
@@ -0,0 +1,115 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.front50.migrations;
+
+import com.netflix.spinnaker.front50.api.model.pipeline.Pipeline;
+import com.netflix.spinnaker.front50.model.pipeline.PipelineDAO;
+import java.time.Clock;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class RedBlackToBlueGreenK8sPipelinesMigration implements Migration {
+
+  private static final LocalDate VALID_UNTIL = LocalDate.of(2023, 6, 1);
+
+  private final PipelineDAO pipelineDAO;
+
+  public RedBlackToBlueGreenK8sPipelinesMigration(PipelineDAO pipelineDAO) {
+    this.pipelineDAO = pipelineDAO;
+  }
+
+  @Override
+  public boolean isValid() {
+    return LocalDate.now(Clock.systemDefaultZone()).isBefore(VALID_UNTIL);
+  }
+
+  @Override
+  public void run() {
+    log.info(
+        "Starting the migration of K8s pipelines using RED/BLACK traffic management strategy to BLUE/GREEN");
+    pipelineDAO.all().stream()
+        .filter(RedBlackToBlueGreenK8sPipelinesMigration.pipelineWithRedBlackStrategyPredicate())
+        .forEach(this::migrate);
+  }
+
+  private void migrate(Pipeline pipeline) {
+    List<Map<String, Object>> stages = new ArrayList<>(pipeline.getStages());
+
+    stages.stream()
+        .filter(RedBlackToBlueGreenK8sPipelinesMigration::kubernetesProvider)
+        .filter(RedBlackToBlueGreenK8sPipelinesMigration::deployManifestStage)
+        .map(RedBlackToBlueGreenK8sPipelinesMigration::getTrafficManagement)
+        .filter(RedBlackToBlueGreenK8sPipelinesMigration::trafficManagementEnabled)
+        .map(RedBlackToBlueGreenK8sPipelinesMigration::getTrafficManagementOptions)
+        .forEach(
+            trafficManagementOptions -> {
+              Object strategy = trafficManagementOptions.get("strategy");
+              if ("redblack".equals(strategy)) {
+                trafficManagementOptions.put("strategy", "bluegreen");
+              }
+            });
+
+    this.pipelineDAO.update(pipeline.getId(), pipeline);
+    log.info(
+        "Migrated pipeline (application: {}, pipelineId: {}, name: {})",
+        pipeline.getApplication(),
+        pipeline.getId(),
+        pipeline.getName());
+  }
+
+  private static Predicate<Pipeline> pipelineWithRedBlackStrategyPredicate() {
+    return pipeline ->
+        pipeline.getStages().stream()
+            .filter(RedBlackToBlueGreenK8sPipelinesMigration::kubernetesProvider)
```

Comment:
Ok for now, long term would want AWS as well as any others where the strategy is red/black to keep terminology consistent across spinnaker.